google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/it/ReadIT.java

```diff
@@ -308,6 +322,94 @@ public void reversed() {
         .inOrder();
   }
 
+  @Test
+  public void reversedWithForcedResumption() throws IOException, InterruptedException {
+    assume()
+        .withMessage("reverse scans are not supported in the emulator")
+        .that(testEnvRule.env())
+        .isNotInstanceOf(EmulatorEnv.class);
+
+    BigtableDataClient client = testEnvRule.env().getDataClient();
+    String tableId = testEnvRule.env().getTableId();
+    String familyId = testEnvRule.env().getFamilyId();
+    String uniqueKey = prefix + "-rev-queries2";
+
+    // Add enough rows that ensures resumption logic is forced
+    Random random;
+    List<Row> expectedResults;
+    try (Batcher<RowMutationEntry, Void> batcher = client.newBulkMutationBatcher(tableId)) {
+
+      byte[] valueBytes = new byte[1024];
+      random = new Random();
+
+      expectedResults = new ArrayList<>();
+
+      for (int i = 0; i < 2 * 1024; i++) {
+        ByteString key = ByteString.copyFromUtf8(String.format("%s-%05d", uniqueKey, i));
+        ByteString qualifier = ByteString.copyFromUtf8("q");
+        long timestamp = System.currentTimeMillis() * 1000;
+        random.nextBytes(valueBytes);
+        ByteString value = ByteString.copyFrom(valueBytes);
+
+        batcher.add(RowMutationEntry.create(key).setCell(familyId, qualifier, timestamp, value));
+        expectedResults.add(
+            Row.create(
+                key,
+                ImmutableList.of(
+                    RowCell.create(familyId, qualifier, timestamp, ImmutableList.of(), value))));
+      }
+    }
+    Collections.reverse(expectedResults);
+
+    BigtableDataSettings.Builder settingsBuilder =
+        testEnvRule.env().getDataClientSettings().toBuilder();
+
+    settingsBuilder.stubSettings().readRowsSettings().retrySettings()
+            .setInitialRpcTimeout(Duration.ofMillis(10))
```

Comment:
If this is 10 wouldn't it guaranteed to retry? do we still need to set the random deadline with interceptor?